### PR TITLE
Refactor: Replace interp2d with interpn

### PR DESCRIPTION
**Pull request by lettlini**
This PR replaces all calls to `interp2d` with `interpn`. 

When conducting a review, unfortunately, caution is required in some places, as not all situations are covered by unit tests.

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [ ] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [ ] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [ ] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [ ] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Review by snilsn — COMMENTED** [review]
Looks good overall @lettlini and does improve readability.

I realized that my first comment is a general question, why did you swap the coordinates?

File: tobac/utils/general.py [hunk 1]
```diff
@@ -91,56 +91,56 @@ def add_coordinates(t, variable_cube):
         # interpolate 2D coordinates:
         elif variable_cube.coord(coord).ndim == 2:
             if variable_cube.coord_dims(coord) == (hdim_1, hdim_2):
-                f = interp2d(dimvec_2, dimvec_1, variable_cube.coord(coord).points)
-                coordinate_points = np.asarray(
-                    [f(a, b) for a, b in zip(t["hdim_2"], t["hdim_1"])]
-                )
+                points = (dimvec_1, dimvec_2)
```
[the remark this answers is not in the document]
Comment (lettlini) [reply tobac/utils/general.py:94]: I think this was necessary for the tuple to have the correct order for `interpn`. I suspect that `interpn` and `interp2d` handle point coordinates differently (there is a difference between coordinates in the Euclidean plane and "image coordinates").

File: tobac/utils/general.py [hunk 2]
```diff
@@ -276,10 +276,10 @@ def add_coordinates_3D(
             hdim2_pos = 1 if time_pos == 2 else 2
             first_dim = coord_to_ax[variable_cube.coord_dims(coord)[hdim2_pos]]
             second_dim = coord_to_ax[variable_cube.coord_dims(coord)[hdim1_pos]]
-            f = interp2d(first_dim[0], second_dim[0], var_coord.points)
-            coordinate_points = [
-                f(a, b) for a, b in zip(t[first_dim[1]], t[second_dim[1]])
-            ]
+            points = (second_dim[0], second_dim[1])
```
[the remark this answers is not in the document]
Comment (lettlini) [reply tobac/utils/general.py]: That was a mistake on my part, thanks for pointing it out!

File: tobac/utils/general.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (w-k-jones) [reply tobac/utils/general.py:94]: Yep, `interp2d` has coordinates in `x`, `y`, whereas `interpn` has coordinates in `i`, `j`... So the order is reversed for the 2d case. It's similar to the `numpy.meshgrid` `indexing="xy"` vs `indexing="ij"` options. Good catch!

[the remark this answers is not in the document]
Comment (snilsn) [reply tobac/utils/general.py:94]: Ok, I see. Thanks for the clarification

**Review by snilsn — APPROVED** [review]
Approving now after the small fix!

**Review by freemansw1 — APPROVED** [review]
I tested this with a fairly large dataset and got the same answers as with the original. Happy for this to be merged. Thanks, @lettlini .

**Comment by freemansw1** [comment]
@lettlini are you happy for this to be merged? If so, I'll go ahead and get it in. One step closer to v1.5 release.

**Comment by lettlini** [comment]
Thanks for the reviews @snilsn @w-k-jones @freemansw1. Happy for this to be merged.

**Pull request merged**